src/java.base/share/classes/java/text/DecimalFormatSymbols.java

```diff
@@ -767,7 +763,8 @@ public boolean equals(Object obj) {
             patternSeparator == other.patternSeparator &&
             infinity.equals(other.infinity) &&
             NaN.equals(other.NaN) &&
-            getCurrencySymbol().equals(other.getCurrencySymbol()) && // possible currency init occurs here
+            // Currency fields are lazy. Init via get call to ensure non-null
+            getCurrencySymbol().equals(other.getCurrencySymbol()) &&
```

Comment:
Probably the same comment refinement can be applied to the location in `hashCode()`.

Reply:
This and the other comments addressed via https://github.com/openjdk/jdk/pull/21315/commits/d31cb2ebc5eb92978bed4c6f38f1ddfd7ff5452c. Thanks!